lib/openstack/openstack_handle/handle.rb

```diff
@@ -28,12 +28,16 @@ class Handle
 
     # Tries both non-SSL and SSL connections to Openstack
     def self.try_connection
-      # attempt to connect without SSL
-      yield "http", {}
-    rescue Excon::Errors::SocketError => err
-      raise unless err.message.include?("end of file reached (EOFError)")
-      # attempt the same connection with SSL
+      # attempt to connect with SSL
       yield "https", {:ssl_verify_peer => false}
+    rescue Excon::Errors::SocketError => err
+      # TODO recognizing something in exception message is not very reliable. But somebody would need to go to excon gem
+      # and do proper exceptions like Excon::Errors::SocketError::UnknownProtocolSSL,
+      # Excon::Errors::SocketError::BadCertificate, etc. all of them inheriting from Excon::Errors::SocketError
```

Comment:
So are you saying you'd have to handle each one individually?  Or can you do something like `rescue (Excon::Error::SocketError.subclasses - [ClassThatShouldNotBeCaught]) => err`